lms/lib/courseware_search/lms_filter_generator.py

```diff
@@ -3,13 +3,37 @@
     * Filter by all courses in which the user is enrolled in
 """
 from student.models import CourseEnrollment
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
 
 from search.filter_generator import SearchFilterGenerator
+from openedx.core.djangoapps.course_groups.cohorts import get_cohort
+from openedx.core.djangoapps.course_groups.models import CourseUserGroupPartitionGroup
 
 
 class LmsSearchFilterGenerator(SearchFilterGenerator):
     """ SearchFilterGenerator for LMS Search """
 
+    def filter_dictionary(self, **kwargs):
+        """ base implementation which filters via start_date """
+        filter_dictionary = super(LmsSearchFilterGenerator, self).filter_dictionary(**kwargs)
+        if 'user' in kwargs and 'course_id' in kwargs and kwargs['course_id']:
+            try:
+                course_key = CourseKey.from_string(kwargs['course_id'])
+            except InvalidKeyError:
+                course_key = SlashSeparatedCourseKey.from_deprecated_string(kwargs['course_id'])
+            cohort = get_cohort(kwargs['user'], course_key, assign=False)
+            partition_group = None
+            try:
+                partition_group = CourseUserGroupPartitionGroup.objects.get(course_user_group=cohort)
+            except CourseUserGroupPartitionGroup.DoesNotExist:
+                pass
+            if partition_group:
+                partition_group = partition_group.group_id
+            filter_dictionary['content_groups'] = unicode(partition_group)
```

Comment:
If partition_group is `None`, then the filter_dictionary gets an entry for `{'content_groups' : u'None'}` which is not going to do the right thing.